### PR TITLE
Added 2-phase fluid properties to take 1-phase fluid properties as inputs

### PR DESCRIPTION
**Pull request by joshuahansel**

### CODE REVIEW

File: modules/fluid_properties/src/userobjects/TwoPhaseFluidProperties.C [hunk 1]
```diff
@@ -15,30 +15,42 @@ InputParameters
...
-               "name '",
-               _vapor_name,
-               "', but a single-phase fluid properties ",
-               "object with this name already exists.");
+  // If the single-phase fluid properties user object names are not provided, it
+  // is implied that these objects will be created by a derived class. In this
+  // case, we need to check that these user objects do not already exist.
+  if (!isParamValid("fp_liquid"))
+    if (_fe_problem.hasUserObject(_liquid_name))
+      mooseError("The two-phase fluid properties object '" + name() + "' is ",
```
Comment (andrsd) [inline modules/fluid_properties/src/userobjects/TwoPhaseFluidProperties.C]: `paramError` should work for this and below.

File: modules/fluid_properties/src/userobjects/TwoPhaseFluidPropertiesIndependent.C [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  return params;
+}
+
+TwoPhaseFluidPropertiesIndependent::TwoPhaseFluidPropertiesIndependent(
+    const InputParameters & parameters)
+  : TwoPhaseFluidProperties(parameters)
+{
+  _fp_liquid = &_fe_problem.getUserObject<SinglePhaseFluidProperties>(_liquid_name);
+  _fp_vapor = &_fe_problem.getUserObject<SinglePhaseFluidProperties>(_vapor_name);
```
Comment (andrsd) [inline modules/fluid_properties/src/userobjects/TwoPhaseFluidPropertiesIndependent.C]: This should be possible in the c-tor initializer list.

`GeneralUserObject`s now inherit from `UserObjectInterface`, so it should be possible to use just `getUserObject` and `hasUserObject` (which is used in `TwoPhaseFluidProperties`).

File: modules/fluid_properties/test/include/auxkernels/TwoPhaseAverageDensityAux.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+protected:
+  virtual Real computeValue() override;
+
+  const VariableValue & _p;
+  const VariableValue & _T;
+
+  const TwoPhaseFluidProperties & _fp_2phase;
+  const SinglePhaseFluidProperties & _fp_liquid;
+  const SinglePhaseFluidProperties & _fp_vapor;
```
Comment (andrsd) [inline modules/fluid_properties/test/include/auxkernels/TwoPhaseAverageDensityAux.h]: missing doxygen doco

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/fluid_properties/src/userobjects/TwoPhaseFluidPropertiesIndependent.C (hunk 2 above)
Comment (joshuahansel) [reply modules/fluid_properties/src/userobjects/TwoPhaseFluidPropertiesIndependent.C]: Regarding the constructor, I don't think I can do this - they have already been initialized in the base class. I tried it, and I got the following error:
```
member initializer '_fp_liquid' does not name a non-static data member or base class
```

**Review by andrsd — APPROVED** [review]

**Pull request merged**